server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/VacationMailetTest.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class VacationMailetTest {
+
+    public static final ZonedDateTime DATE_TIME_1 = ZonedDateTime.parse("2016-10-09T08:07:06+07:00[Asia/Vientiane]");
+    public static final ZonedDateTime DATE_TIME_2 = ZonedDateTime.parse("2017-10-09T08:07:06+07:00[Asia/Vientiane]");
+    public static final ZonedDateTime DATE_TIME_3 = ZonedDateTime.parse("2018-10-09T08:07:06+07:00[Asia/Vientiane]");
+
+    public static final String USERNAME = "benwa@apache.org";
+    private VacationMailet testee;
+    private VacationRepository vacationRepository;
+    private ZonedDateTimeProvider zonedDateTimeProvider;
+    private FakeMailContext fakeMailContext;
+    private MailAddress originalSender;
+    private MailAddress originalRecipient;
+
+    @Before
+    public void setUp() throws Exception {
+        this.originalSender = new MailAddress("distant@apache.org");
+        this.originalRecipient = new MailAddress(USERNAME);
+
+        vacationRepository = mock(VacationRepository.class);
+        zonedDateTimeProvider = mock(ZonedDateTimeProvider.class);
+        testee = new VacationMailet(vacationRepository, zonedDateTimeProvider);
+        fakeMailContext = new FakeMailContext();
+        testee.init(new FakeMailetConfig("vacation", fakeMailContext));
+    }
+
+    @Test
+    public void unactivatedVacationShouldNotSendNotification() throws Exception {
+        FakeMail mail = createFakeMail();
+        when(zonedDateTimeProvider.get()).thenReturn(DATE_TIME_2);
+        when(vacationRepository.retrieveVacation(AccountId.create(USERNAME)))
+            .thenReturn(CompletableFuture.completedFuture(VacationRepository.DEFAULT_VACATION));
+
+        testee.service(mail);
+
+        assertThat(fakeMailContext.getSentMails()).isEmpty();
+    }
+
+    @Test
+    public void activateVacationShouldSendNotification() throws Exception {
+        FakeMail mail = createFakeMail();
+        when(vacationRepository.retrieveVacation(AccountId.create(USERNAME)))
+            .thenReturn(CompletableFuture.completedFuture(
+                Vacation.builder()
+                    .enabled(true)
+                    .fromDate(Optional.of(DATE_TIME_1))
+                    .toDate(Optional.of(DATE_TIME_3))
+                    .textBody("Explaining my vacation")
+                    .build()));
+        when(zonedDateTimeProvider.get()).thenReturn(DATE_TIME_2);
+
+        testee.service(mail);
+
+        FakeMailContext.SentMail expected = new FakeMailContext.SentMail(originalRecipient, ImmutableList.of(originalSender), null);
+        assertThat(fakeMailContext.getSentMails()).containsExactly(expected);
+    }
+
+    private FakeMail createFakeMail() throws MessagingException {
+        FakeMail mail = new FakeMail();
```

Comment:
You should not have to write this createFakeMail method. You do it because FakeMail doesn't have a great API. Please make FakeMail better and drop this method, please.